### ISSUE TITLE
Resolve coersion to the base field problem in sage

### ISSUE THREAD

**Issue by drskalman** [w3f/algebraic-torus#1]
We need to define an intermediary (homomorphism)map to avoid the following Type Error:
```
sage: gamma(1,1,1)
92*a + 55
sage:  a in Fqsqrt
True
sage: a == (gamma(1,1,1) - 55)/92
True
sage: Fqsqrt(a)
a
sage: Fqsqrt((gamma(1,1,1) - 55)/92)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-108-594d8daadf7c> in <module>
----> 1 Fqsqrt((gamma(Integer(1),Integer(1),Integer(1)) - Integer(55))/Integer(92))

~/code/sage/local/lib/python3.9/site-packages/sage/structure/parent.pyx in sage.structure.parent.Parent.__call__ (build/cythonized/sage/structure/parent.c:9335)()
    896         if mor is not None:
    897             if no_extra_args:
--> 898                 return mor._call_(x)
    899             else:
    900                 return mor._call_with_args(x, args, kwds)

~/code/sage/local/lib/python3.9/site-packages/sage/structure/coerce_maps.pyx in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (build/cythonized/sage/structure/coerce_maps.c:4622)()
    159                 print(type(C), C)
    160                 print(type(C._element_constructor), C._element_constructor)
--> 161             raise
    162
    163     cpdef Element _call_with_args(self, x, args=(), kwds={}):

~/code/sage/local/lib/python3.9/site-packages/sage/structure/coerce_maps.pyx in sage.structure.coerce_maps.DefaultConvertMap_unique._call_ (build/cythonized/sage/structure/coerce_maps.c:4514)()
    154         cdef Parent C = self._codomain
    155         try:
--> 156             return C._element_constructor(x)
    157         except Exception:
    158             if print_warnings:

~/code/sage/local/lib/python3.9/site-packages/sage/rings/finite_rings/finite_field_givaro.py in _element_constructor_(self, e)
    367             2*a4^3 + 2*a4^2 + 1
    368         """                                                                                                                                                                                                                          
--> 369         return self._cache.element_from_data(e)
    370
    371     def gen(self, n=0):

~/code/sage/local/lib/python3.9/site-packages/sage/rings/finite_rings/element_givaro.pyx in sage.rings.finite_rings.element_givaro.Cache_givaro.element_from_data (build/cythonized/sage/rings/finite_rings/element_givaro.cpp:7573)()
    312         return make_FiniteField_givaroElement(self, res)
    313
--> 314     cpdef FiniteField_givaroElement element_from_data(self, e):
    315         """                                                                                                                                                                                                                          
    316         Coerces several data types to ``self``.

~/code/sage/local/lib/python3.9/site-packages/sage/rings/finite_rings/element_givaro.pyx in sage.rings.finite_rings.element_givaro.Cache_givaro.element_from_data (build/cythonized/sage/rings/finite_rings/element_givaro.cpp:7155)()
    448
    449         else:
--> 450             raise TypeError("unable to coerce %r" % type(e))                                                                                                                                                                         
    451
    452         cdef GEN t

TypeError: unable to coerce <class 'sage.rings.polynomial.polynomial_quotient_ring.PolynomialQuotientRing_field_with_category.element_class'>
```

**Comment by drskalman** [comment]
This mainly resolved by 549ebae. The complete computation of rationalization maps requires resolution of #6.

**Comment by drskalman** [comment]
I need to implement the inverse the same way because the sage inverse_image does not give an homomorphism object which we need it for some reason.
